### PR TITLE
Add button interactions

### PR DESCRIPTION
**Pull request by Chris-Maina**
##### What does this PR do?
Implements feedback to add button interactions, modify font styles of search input placeholder and society points

##### Description of Task to be completed?
* Add cursor pointer to buttons
* Add box shadow and transition to hover state of buttons
* Add bold font weight to society stats
* Refactor font family of search input placeholder
* Remove border radius on progress bar

### CODE REVIEW

File: src/app/Dashboard/styles/dashboard.scss [hunk 1]
```diff
@@ -78,15 +78,14 @@
   .button__add {
     height: rem(32px);
     width: rem(143px);
-    background-color: #F5A623;
-    box-shadow: inset 0 1px 3px 0 rgba(0,0,0,0.5), 0 12px 35px 0 rgba(52,77,178,0.34);
+    background-color: $orange;
```
Comment (kimani-ndegwa) [inline src/app/Dashboard/styles/dashboard.scss]: Do we really have orange in the brand guideline?

Comment (Chris-Maina) [reply src/app/Dashboard/styles/dashboard.scss]: No, we do not. I have changed it to Andela Gold.

File: src/app/Dashboard/components/DashboardComponent.js [hunk 2]
```diff
@@ -83,11 +83,11 @@ export class DashboardComponent extends Component {
                 <div className='user-dashboard__actions col-sm-12'>
                   <h3 className='user-dashboard__title'>My Activities</h3>
                   <div>
-                    <ButtonComponent className='button__add'>
+                    <ButtonComponent className='user-dashboard__button button__add'>
```
Comment (Kachulio1) [inline src/app/Dashboard/components/DashboardComponent.js]: Can we refactor this component, we can have the span(s) inside the Button component and have the name passed as props and Icon. but we can do that in another PR.
And change the name of the component to Button

Comment (Kachulio1) [reply src/app/Dashboard/components/DashboardComponent.js]: The reason is

File: src/app/common/styles/sidebar.scss [hunk 3]
```diff
@@ -44,20 +45,35 @@
...
+  height: rem(14.45px);
+  width: rem(24.51px);
+
   &.dashboard {
-    color: #FFFFFF;
     background-image: url(../../../images/dashboard.svg);
   }
+
   &.istelle {
+    border: rem(1.5px) solid $lightAccent;
```
Comment (Kachulio1) [inline src/app/common/styles/sidebar.scss]: Let's use a Mixin since all the properties for the society logo selector are the same.
We can avoid writing the same code over and over again.
